Rakefile

```diff
@@ -86,3 +86,190 @@ EOM
+  end
+
+  # TODO: This should be refactored to use the https://github.com/puppetlabs/puppetlabs-peadm
+  # module for PE setup
+  desc 'Sets up PE on the master'
+  task :setup_pe do
+    on_master { |master| master.bolt_run_script('spec/support/acceptance/install_pe.sh') }
+  end
+
+  desc 'Sets up the ServiceNow instance'
+  task :setup_servicenow_instance do
+    # Start the mock ServiceNow instance. Note that the script's already idempotent
+    # (and fast enough) so we don't need to check the inventory file for the created
+    # ServiceNow instance
+    on_master do |master|
+      puts("Starting the mock ServiceNow instance at the master (#{master.uri})")
+      master.bolt_upload_file('./spec/support/acceptance/servicenow', '/tmp/servicenow')
+      master.bolt_run_script('spec/support/acceptance/start_mock_servicenow_instance.sh')
+    end
+
+    # Update the inventory file
+    puts('Updating the inventory.yaml file with the mock ServiceNow instance credentials')
+    inventory_hash = LitmusHelpers.inventory_hash_from_inventory_file
+    servicenow_group = inventory_hash['groups'].find { |g| g['name'] =~ %r{servicenow} }
+    unless servicenow_group
+      servicenow_group = { 'name' => 'servicenow_nodes' }
+      inventory_hash['groups'].push(servicenow_group)
+    end
+    servicenow_group['targets'] = [{
+      'uri' => "#{on_master(&:uri)}:1080",
+      'config' => {
+        'transport' => 'remote',
+        'remote' => {
+          'user' => 'mock_user',
+          'password' => 'mock_password'
+        }
+      },
+      'vars' => {
+        'is_servicenow_instance' => true
+      }
+    }]
+    write_to_inventory_file(inventory_hash, 'inventory.yaml')
+  end
+
+  desc 'Sets up the ServiceNow CMDB with entries for each VM'
+  task :setup_servicenow_cmdb do
+    if sys_id = CMDBHelpers.record_ids[on_master(&:uri)]
+      # Double-check that the CMDB record exists on the ServiceNow instance
+      begin
+        CMDBHelpers.get_record(sys_id)
+        puts("A CMDB record's already been created for the master (sys_id = #{sys_id})")
+        next
+      rescue CMDBRecordRetrievalError => e
+        raise "inventory.yaml reports a sys_id of #{sys_id} for master's CMDB record, but servicenow_tasks::get_record fails to retrieve the record: #{e}"
+      end
+    end
+
+    # CMDB record doesn't exist so create it
+    puts("Creating the master's CMDB record ...")
+    fields_template = JSON.parse(File.read('spec/support/acceptance/cmdb_record_template.json'))
+    fields = CMDBHelpers.create_record(fields_template.merge('fqdn' => on_master(&:uri)))
+    sys_id = fields['sys_id']
+
+    # Add the CMDB record to the inventory.yaml file
+    puts("Adding the created CMDB record (sys_id = #{sys_id}) to inventory['vars'] ...")
+    master_uri = on_master(&:uri)
+    inventory_hash = inventory_hash_from_inventory_file
+    inventory_hash['vars'] = {
+      'cmdb_record_ids' => CMDBHelpers.record_ids.merge(master_uri => sys_id),
```

Comment:
This shouldn't be necessary once `servicenow_tasks` has a `get_records` task b/c I can just query for CMDB entry where `fqdn == uri`.